sdk/typescript/.changes/unreleased/Changed-20240806-223717.yaml

```diff
@@ -0,0 +1,7 @@
+kind: Changed
+body: |
+  Drop CommonJS support in favor of Module. May be a breaking change if you use very old modules.
```

Comment:
I am wondering:
1. What is a "very old module"?
2. Is there any way that can make this change without breaking "very old modules"?
3. How many "very old modules" are available on https://daggerverse.dev? We may need @grouville @jpadams @vito @marcosnils help to answer this.

Reply:
I think we should very much try and do 2. If we do go forward with this specific change, we need to only apply it based on the contents of the `engineVersion` - if it's larger than "vX.Y.Z", we can set this, otherwise, we don't.

Can you give an example of a module that *would* break?

Reply:
> Can we make this change without breaking "very old modules"?

Sadly not, except if we use different `graphql-request` version depending on the module's type, but that seems hacky.

I don't have an example in mind, there's not much packages that use commonJS anymore.

According to this [issue](https://github.com/nodejs/node/issues/33222), it doesn't look backward compatible but again, I don't know